### ISSUE TITLE
Fedora ecryptfs-utils

### ISSUE THREAD

**Issue by rapgro** [mhogomchungu/ecryptfs-simple#3]
We've an issue with OpenSSL 1.1.x, maybe you could help to fix ecryptfs-utils. Upstreamed patch does not work:
https://bugzilla.redhat.com/show_bug.cgi?id=1384023

**Comment by mhogomchungu** [comment]
Will look into it later on today.

While here, i planned to release a new version of SiriKali today but i will wait until this[1] issue on OSX is resolved.

Your cmake options should have "-DUSE_POLKIT=true" option and a complete cmake command could be something like:
```
 cmake -DCMAKE_INSTALL_PREFIX=/usr -DQT5=true -DUSE_POLKIT=true -DCMAKE_BUILD_TYPE=release . ..
```
The polkit option will cause a polkit prompt to be generated every time the application is started and those who do not want the prompt because they do not use ecryptfs-simple backend can turn off polkit support by changing a config option named "ElevatePrivileges" to false. The config file should be at "~/.config/SiriKali/SiriKali.conf".

I have tested it in fedora 25 so i know it works but a confirmation will be appreciated.

You should also build zuluCrypt with the same "-DUSE_POLKIT=true" cmake option when the new version is released.

Both these projects elevates privileges differently now and in a way that keeps GUI components on the normal user side of privileges. The code that needs elevated privileges now sits on a background process called "sirikali.pkexec" for SiriKali and "zuluPolkit" for zuluCrypt.

[1] https://github.com/mhogomchungu/sirikali/issues/21

**Comment by rapgro** [comment]
Package review of SiriKali is stalled till ecryptfs-utils is fixed.

**Comment by mhogomchungu** [comment]
I cant seem to be able to reproduce the error.

Tested with openssl 1.1.0e and ecryptfs from here: https://code.launchpad.net/~jelle-vdwaa/ecryptfs/ecryptfs

**Comment by rapgro** [comment]
It's fixed at upstream with a PR.

**Comment by mhogomchungu** [comment]
You will need this[1] patch to make released version of SiriKali(1.2.7) work with suid-root-less ecryptfs-simple in fedora. This is because Fedora does not have "su" binary where SiriKali expects it to be.

In SiriKali 1.2.7 going forward, polkit support can be set from GUI by selecting ```menu->enable/disable polkit support```.

[1] https://github.com/mhogomchungu/sirikali/commit/4ebe69f0dd4ddc3f57764aa697f39a9ae305cae6
